### PR TITLE
add module for adding passwords

### PR DESCRIPTION
**Pull request by elwood218**
## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Passwords cannot be added.

## What is the new behavior?

- Passwords can be added or deleted to/out of the password store.
- If passwords exists nothing will be done.

## Other information

If I would knew how to read passwords if I am logged in on the Checkmk server then it would be possible to compare passwords and change them. For now it only would be possible to create a password and could think about to force password set to overwrite it nevertheless if it differs or not.
The API lets only set "editable by" to a contact group the editor is part of. So as user "automation" I can only set "admin" if automation would be Administrator.

### CODE REVIEW

**Review by lgetwan — CHANGES_REQUESTED** [review]
I successfully tested your new module. Thanks for contributing!

I like Your idea to provide a "force new password". What do you think about doing it like this:

- present: creates a pw, if absent and doesn't do anything if the pw is already present.
- absent: deletes a pw if existing
- reset: resets the "password" value if the pw exists. If the pw doesn't exist, it exits with "failed"

**Comment by elwood218** [comment]
@lgetwan Hi. What do you mean by "reset"? For me reset more sounds like setting a changed password back to the previous. 
Btw is there a way to read password on cli?

**Comment by lgetwan** [comment]
@elwood218: There might be better words than "reset", true.
I guess, there _is_ a way to decrypt the PW store on the CLI. I haven't tried, yet. Maybe, having a close look at ~/lib/check_mk/utils/password_store.py and encryption.py helps.Also checkout https://checkmk.com/werk/13633

**Comment by elwood218** [comment]
@lgetwan Yes I already had a look into both you mentioned. Thought or hoped there is a CLI command for that. But then first let us implement like how it is now and later think of the other way.

File: plugins/modules/password.py [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+description: Manage passwords in the Checkmk password store.
+
+options:
+    name:
+        description:
+            - Name/title of the password.
+        required: true
+        type: str
+        aliases: [title]
+    ident:
```
Comment (robin-tribe29) [inline plugins/modules/password.py:28]: In the `tag_group` module we use "id". I think we should do that here too.
I am aware, that the API calls it "ident", but I think "id" is better understood.

File: plugins/modules/password.py [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+# Delete a password.
+- name: "Delete a password."
+  tribe29.checkmk.host:
+    server_url: "http://localhost/"
+    site: "my_site"
+    automation_user: "automation"
+    automation_secret: "$SECRET"
+    name: "my-password"
+    state: "absent"
+"""
```
Comment (robin-tribe29) [inline plugins/modules/password.py:78]: I think in both examples the "password" key is missing, right?

**Review by robin-tribe29 — COMMENTED** [review]
Is there a reason, that you do not deal with the "shared" key from the API?

**Comment by lgetwan** [comment]
Ok, my wording "reset password" was probably a bad choice. What I meant:

1. state = "absent": Delete password entry if existing.
2. state = "present": Create password entry if not existing. If existing, do NOT touch the actual password, even if the parameter is set.
3. state = "reset": Create password entry if not existing. If existing, set the actual password to the value of the password parameter.

What wording would you suggest instead of "reset"? "overwrite" or "force_new_password" might be alternatives, but I think, I'm not on the peak of my creativity, today. :-p

**Comment by elwood218** [comment]
@lgetwan I would say "force" or "overwrite" because "force" for example is also used in other similar Ansible modules if I remember correct.

**Comment by robin-tribe29** [comment]
Is there a perceivable use case, where you want to use "present" to ensure a password is there without changing anything? I feel like the answer here is "no". So I would expect, that "present" ensures the password I want to set is actually set, without caring about a potentially existing password.

I feel like "reset" is still more confusing, although I get your idea @lgetwan. But I also feel like "force" should not be necessary. If there is a use case for my former example, then I think we should go with "present", "absent" and "force".

P.S.: @elwood218 you marked my comments as resolved, but I see no new commits, nor a comment from you. Do you just need to push the changes?

**Pull request closed without merging**